### DISCUSSION TITLE
Splitting a Single Git Repo into Multiple Projects on Vercel

### DISCUSSION THREAD

**Discussion by laurent-le-graverend** [vercel/community#5380] [Feature Question] [answered]
### Question

I'm managing a Git repository that contains multiple branches, where each branch represents a different version of a web application. These branches function as distinct projects.

I'm looking to achieve the following setup:
- Project 1 linked to Branch A of the GitHub repository
- Project 2 linked to Branch B of the GitHub repository

This setup is straightforward on Netlify, but I'm encountering difficulties replicating it on Vercel. Is there a way to configure two separate projects on Vercel, each linked to a specific branch of the same GitHub repository?

**Comment by versecafe** [comment, marked as answer]
## Option 1
Using Vercel's custom production branch setting. While not exactly efficient as builds will call on both projects always putting one in the build que, you should be able to make two copies of the project and set their production branches to your target branch following this document https://vercel.com/blog/custom-production-branch.

## Option 2
Configuring a custom CI. If desired you can configure a github CI linked to two different projects by having a different `.vercel` folder in each branch configured to different projects in Vercel and then in CI call `vercel --prod` after configuring the Vercel CLI for your given CI platform. This will avoid the double build and que issue of option 1 but is far more work to configure.

**Reply by amyegan** [reply]
This is a great answer! For anyone choosing option 1, I recommend also looking at the [Ignored Build Step](https://vercel.com/docs/projects/overview#ignored-build-step) feature to limit unnecessary deployments from branches associated with the other project.

**Reply by laurent-le-graverend** [reply]
Thanks for the suggestion! I went with option 1 and it's working. But yeah, it's not really ideal to queue up unrelated branches and then just cancel them at the last minute automatically. Maybe this could be improved this by figuring out a way to exclude or include certain branches beforehand, for a future update!
